app/navigators/ChargeENavigator.scala

```diff
@@ -116,14 +99,67 @@ class ChargeENavigator @Inject()(val dataCacheConnector: UserAnswersCacheConnect
     case AddMembersPage => addMembers(ua, srn, startDate, accessType, version)
     case DeleteMemberPage => deleteMemberRoutes(ua, srn, startDate, accessType, version)
   }
 
+  private def routeFromIsPublicServicePensionsRemedyPage(userAnswers: UserAnswers, srn: String, startDate: LocalDate
+                                                           , accessType: AccessType, version: Int, index: Int): Call = {
+    userAnswers.get(IsPublicServicePensionsRemedyPage(ChargeTypeAnnualAllowance, index)) match {
+      case Some(true) => controllers.mccloud.routes.IsChargeInAdditionReportedController
+        .onPageLoad(ChargeTypeAnnualAllowance, NormalMode, srn, startDate, accessType, version, index)
+      case Some(false) => CheckYourAnswersController.onPageLoad(srn, startDate, accessType, version, index)
+      case _ => sessionExpiredPage
+    }
+  }
+
+  private def routeFromIsChargeInAdditionReportedPage(userAnswers: UserAnswers, srn: String, startDate: LocalDate
+                                                      , accessType: AccessType, version: Int, index: Int): Call = {
+    userAnswers.get(IsChargeInAdditionReportedPage(ChargeTypeAnnualAllowance, index)) match {
+      case Some(true) => controllers.mccloud.routes.WasAnotherPensionSchemeController
+        .onPageLoad(ChargeTypeAnnualAllowance, NormalMode, srn, startDate, accessType, version, index)
+      case Some(false) => CheckYourAnswersController.onPageLoad(srn, startDate, accessType, version, index)
+      case _ => sessionExpiredPage
+    }
+  }
+
+  private def routeFromWasAnotherPensionSchemePage(userAnswers: UserAnswers, srn: String, startDate: LocalDate
+                                                   , accessType: AccessType, version: Int, index: Int): Call = {
+    userAnswers.get(WasAnotherPensionSchemePage(ChargeTypeAnnualAllowance, index)) match {
+      case Some(true) => controllers.mccloud.routes.EnterPstrController
+        .onPageLoad(ChargeTypeAnnualAllowance, NormalMode, srn, startDate, accessType, version, index, countSchemeSize(userAnswers, index))
+      case Some(false) =>
+        controllers.mccloud.routes.TaxYearReportedAndPaidController
+          .onPageLoad(ChargeTypeAnnualAllowance, NormalMode, srn, startDate, accessType, version, index, None)
+      case _ => sessionExpiredPage
+    }
+  }
+
+  private def routeFromChargeAmountReportedPage(userAnswers: UserAnswers, srn: String, startDate: LocalDate, accessType: AccessType
+                                                , version: Int, index: Int, schemeIndex: Option[Int]): Call = {
+    val schemeSize = countSchemeSize(userAnswers, index)
+    val schemeSizeLessThan5 = schemeSize > 0 && schemeSize < 5
+    (schemeIndex, schemeSizeLessThan5) match {
+      case (Some(i), true) => controllers.mccloud.routes.AddAnotherPensionSchemeController
+        .onPageLoad(ChargeTypeAnnualAllowance, NormalMode, srn, startDate, accessType, version, index, i)
```

Comment:
same as above

Reply:
Fixed, added 2 more cases to handle None and case _ , _